packages/expo/src/index.ts

```diff
@@ -0,0 +1,45 @@
+import Constants from "expo-constants";
+
+import { generateReactHelpers } from "@uploadthing/react/native";
+import type * as _TS_FIND_ME_1 from "@uploadthing/shared";
+import type { FileRouter } from "uploadthing/internal/types";
+
+import { GENERATE_useDocumentUploader } from "./document-picker";
+import { GENERATE_useImageUploader } from "./image-picker";
+
+export interface GenerateTypedHelpersOptions {
+  /**
+   * URL to the UploadThing API endpoint
+   * @example "/api/uploadthing"
+   * @example "https://www.example.com/api/uploadthing"
+   *
+   * If relative, host will be inferred from either the `EXPO_PUBLIC_SERVER_ORIGIN` environment variable or `ExpoConstants.hostUri`
+   *
+   * @default (process.env.EXPO_PUBLIC_SERVER_ORIGIN ?? ExpoConstants.debuggerHost) + "/api/uploadthing"
+   */
+  url?: URL | string;
+}
+
+export const generateReactNativeHelpers = <TRouter extends FileRouter>(
+  initOpts?: GenerateTypedHelpersOptions,
+) => {
+  const debuggerHost = Constants.expoConfig?.hostUri;
+  let url: URL;
+  try {
+    url = new URL(
+      initOpts?.url ?? "/api/uploadthing",
+      process.env.EXPO_PUBLIC_SERVER_ORIGIN ?? `http://${debuggerHost}`,
+    );
+    console.log("resolved url", url);
+  } catch (e) {
```

Comment:
cc @EvanBacon - how would you recommend grabbing the absolute url?

I know Expo allows for relative fetch calls (e.g. `fetch("/api/uploadthing")` but we only work with URL objects inside the uploadthing internals so we need to parse the absolute url some way.

if a string is passed in the objects below (L40-L42), UploadThing falls back to using `window.location.host` but that's not available in Expo and will throw runtime errors, we need to pass a URL object

---

tldr; is there some easy way to grab the server url from Expo?